puppet/modules/kubernetes/templates/kubelet.service.erb

```diff
@@ -95,15 +95,27 @@ ExecStart=<%= scope['kubernetes::_dest_dir'] %>/kubelet \
     @cgroup_system_reserved << "memory=#{@cgroup_system_reserved_memory}" unless @cgroup_system_reserved_memory.nil? or @cgroup_system_reserved_memory == 'nil'
     if @cgroup_system_reserved.length > 0
 -%>
-  "--system-reserved=<%= @cgroup_system_reserved.join(',') %>" \
+  --system-reserved=<%= @cgroup_system_reserved.join(',') %> \
 <% end -%>
 <% end -%>
 <% end -%>
 <% if @cert_file and @key_file -%>
-  "--tls-cert-file=<%= @cert_file %>" \
-  "--tls-private-key-file=<%= @key_file %>" \
+  --tls-cert-file=<%= @cert_file %> \
+  --tls-private-key-file=<%= @key_file %> \
 <% end -%>
-  "--eviction-hard=memory.available<<%= @hard_eviction_memory_threshold %>" \
+<%
+    # build eviction hard command line
+    @eviction_hard = []
+    @eviction_hard << "memory.available<#{@hard_eviction_memory_available_threshold}" unless @hard_eviction_memory_available_threshold.nil? or @hard_eviction_memory_available_threshold == 'nil'
+    @eviction_hard << "nodefs.available<#{@hard_eviction_nodefs_available_threshold}" unless @hard_eviction_nodefs_available_threshold.nil? or @hard_eviction_nodefs_available_threshold == 'nil'
+    @eviction_hard << "nodefs.inodesFree<#{@hard_eviction_nodefs_inodes_free_threshold}" unless @hard_eviction_nodefs_inodes_free_threshold.nil? or @hard_eviction_nodefs_inodes_free_threshold == 'nil'
+    if @eviction_hard.length > 0
+-%>
+  --eviction-hard=<%= @eviction_hard.join(',') %> \
+<% end -%>
+  --eviction-soft=memory.available<<%= @soft_eviction_memory_available_threshold %> \
+  --eviction-soft-grace-period=memory.available=<%= @soft_eviction_memory_available_grace_period %> \
+  --eviction-max-pod-grace-period=<%= @eviction_max_pod_grace_preriod %> \
   --logtostderr=true
```

Comment:
This PR should be about increasing the reservations as the title says, why are you doing all of this?

Reply:
yeah got a bit into it, was basically putting it here as an example a soft eviction threshold that might be good, but a PR wasn't the right place for it